File: tests/js/closure_scope_deep/mutual_recursion_even_odd.js
// vybe-test: js/closure_scope_deep/mutual_recursion_even_odd
// origin: languages/js/tests/js/test_closure_scope_deep.rs

function __line(...args) {
    // console.log joins its arguments with a single space. String() is the
    // coercion Vybe's logging host applies to each one.
    return args.map(String).join(" ");
}

function __check(got, want) {
    if (got !== want) {
        console.log("FAIL: want [" + want + "] got [" + got + "]");
        throw new Error("assertion failed");
    }
}

function isEven(n) {
    if (n === 0) return true;
    return isOdd(n - 1);
}
function isOdd(n) {
    if (n === 0) return false;
    return isEven(n - 1);
}
__check(__line(isEven(4)), "true");
__check(__line(isOdd(7)), "true");
__check(__line(isEven(1)), "false");

File: tests/lua/functions/mutual_recursion_even_odd.lua
-- vybe-test: lua/functions/mutual_recursion_even_odd
-- origin: languages/lua/tests/lua/test_functions.rs

local __w1 = "true"
local __i = 0

local is_even, is_odd
function is_even(n)
  if n == 0 then return true end
  return is_odd(n - 1)
end
function is_odd(n)
  if n == 0 then return false end
  return is_even(n - 1)
end
do local __t = tostring(is_even(4)); __i = __i + 1
  if __i == 1 and __t ~= __w1 then error("FAIL: want [" .. __w1 .. "] got [" .. __t .. "]") end end

if __i == 0 then error("FAIL: no output, wanted [" .. __w1 .. "]") end
